test/e2e/broker_defaults_webhook_test.go

```diff
@@ -0,0 +1,170 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"knative.dev/eventing/pkg/apis/config"
+	eventingduck "knative.dev/eventing/pkg/apis/duck/v1"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/pkg/reconciler"
+	"knative.dev/pkg/system"
+)
+
+func TestBrokerNamespaceDefaulting(t *testing.T) {
+	ctx := context.Background()
+
+	c := testlib.Setup(t, true)
+	defer testlib.TearDown(c)
+
+	err := reconciler.RetryTestErrors(func(attempt int) error {
+
+		t.Log("Updating defaulting ConfigMap attempt:", attempt)
+
+		cm, err := c.Kube.CoreV1().ConfigMaps(system.Namespace()).Get(ctx, config.DefaultsConfigName, metav1.GetOptions{})
+		assert.Nil(t, err)
+
+		// Preserve existing namespace defaults.
+		defaults := make(map[string]map[string]interface{})
+		err = yaml.Unmarshal([]byte(cm.Data[config.BrokerDefaultsKey]), defaults)
+		assert.Nil(t, err)
+
+		if _, ok := defaults["namespaceDefaults"]; !ok {
+			defaults["namespaceDefaults"] = make(map[string]interface{})
+		}
+
+		defaults["namespaceDefaults"][c.Namespace] = map[string]interface{}{
+			"brokerClass": brokerClass,
+			"delivery": map[string]interface{}{
+				"retry":         5,
+				"backoffPolicy": "exponential",
+				"backoffDelay":  "PT0.5S",
+			},
+		}
+
+		b, err := yaml.Marshal(defaults)
+		assert.Nil(t, err)
+
+		cm.Data[config.BrokerDefaultsKey] = string(b)
+
+		cm, err = c.Kube.CoreV1().ConfigMaps(system.Namespace()).Update(ctx, cm, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+
+		b, err = yaml.Marshal(cm.Data[config.BrokerDefaultsKey])
+		if err != nil {
+			t.Log("error", err)
+		} else {
+			t.Log("CM updated - new values:", string(b))
+		}
+
+		return nil
+	})
+	assert.Nil(t, err)
+
+	// Create a Broker and check whether it has the DeliverySpec set as we specified above.
+	// Since the webhook receives the updates at some undetermined time after the update to reduce flakiness retry after
+	// a delay and check whether it has the shape we want it to have.
+
+	namePrefix := "xyz"
+	lastName := ""
+
+	err = reconciler.RetryTestErrors(func(attempt int) error {
```

Comment:
`RetryTestErrors` doesn't have the same semantic as: https://github.com/knative/eventing/blob/14a9fbbb1a11390ca6ff85dad00bd0d377d1bb34/test/e2e/channel_defaults_webhook_test.go#L116
We want to retry custom errors like `fmt.Errorf("webhook hasn't seen the update: %+v", broker.Spec.Delivery)`.

Reply:
thx, good catch.
Would it be worth it to have another method (eg. RetryTestOrCustomErrors) in pkg ?

Reply:
I'm not sure if this is a shared-use case, do you think it is?